Sources/Testing/Attachments/Test.Attachment.swift

```diff
@@ -0,0 +1,300 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+private import _TestingInternals
+
+@_spi(Experimental)
+extension Test {
+  /// A type describing values that can be attached to the output of a test run
+  /// and inspected later by the user.
+  ///
+  /// Attachments are included in test reports in Xcode or written to disk when
+  /// tests are run at the command line. To create an attachment, you need a
+  /// value of some type that conforms to ``Test/Attachable``. Initialize an
+  /// instance of ``Test/Attachment`` with that value and, optionally, a
+  /// preferred filename to use when writing to disk.
+  public struct Attachment: Sendable {
+    /// The value of this attachment.
+    ///
+    /// The type of this property's value may not match the type of the value
+    /// originally used to create this attachment.
+    public var attachableValue: any Attachable & Sendable /* & Copyable rdar://137614425 */
+
+    /// The source location of the attachment.
+    public var sourceLocation: SourceLocation
+
+    /// The default preferred name to use if the developer does not supply one.
+    package static var defaultPreferredName: String {
```

Comment:
Why `package` for this? I don't see it being used in any other modules

Reply:
We'll be using it in subsequent PRs to add e.g. Foundation extensions.